lib/dolly/collection.rb

```diff
@@ -20,6 +20,15 @@ def last
       to_a.last
     end
 
+    def update_properties! properties ={}
+      properties.each do |key, value|
+        json.gsub! %r{\"#{key}\":\"[^\"]+\"}, "\"#{key}\":\"#{value}\""
```

Comment:
json is the unparsed json string, so instead of parsing then locking for key and values to replace... I use regex, as it will be much faster

on the regex itself:

`\"` is a literral `\"` as the json string have the `"` escaped then I am matching the key of the property, and find the closing `\"` then there is a literal `:` following by the value enclosed in `\"`. The regex finds a `\"` followed by anything not that is what  `[^\"]` does, and we need to be something, that is why `+`, and now that I see this... I think this will fail if the value is empty or `nil`,

I will need to write more tests on this to ensure that the nil and empty values get updated.